tensorflow_io/json/kernels/json_kernels.cc

```diff
@@ -199,5 +205,229 @@ REGISTER_KERNEL_BUILDER(Name("ReadJSON").Device(DEVICE_CPU),
                         ReadJSONOp);
 
 }  // namespace
+
+
+class JSONIndexable : public IOIndexableInterface {
+ public:
+  JSONIndexable(Env* env)
+  : env_(env) {}
+
+  ~JSONIndexable() {}
+  Status Init(const std::vector<string>& input, const std::vector<string>& metadata, const void* memory_data, const int64 memory_size) override {
+    if (input.size() > 1) {
+      return errors::InvalidArgument("more than 1 filename is not supported");
+    }
+    const string& filename = input[0];
+    file_.reset(new SizedRandomAccessFile(env_, filename, memory_data, memory_size));
+    TF_RETURN_IF_ERROR(file_->GetFileSize(&file_size_));
+
+    json_file_.reset(new ArrowRandomAccessFile(file_.get(), file_size_));
+
+    ::arrow::Status status;
+
+    status = ::arrow::json::TableReader::Make(::arrow::default_memory_pool(), json_file_, ::arrow::json::ReadOptions::Defaults(), ::arrow::json::ParseOptions::Defaults(), &reader_);
+    if (!status.ok()) {
+      return errors::InvalidArgument("unable to make a TableReader: ", status);
+    }
+    status = reader_->Read(&table_);
+    if (!status.ok()) {
+      return errors::InvalidArgument("unable to read table: ", status);
+    }
+
+    std::vector<string> columns;
+    for (size_t i = 0; i < metadata.size(); i++) {
+      if (metadata[i].find_first_of("column: ") == 0) {
+        columns.emplace_back(metadata[i].substr(8));
+      }
+    }
+
+    columns_index_.clear();
+    if (columns.size() == 0) {
+      for (int i = 0; i < table_->num_columns(); i++) {
+        columns_index_.push_back(i);
+      }
+    } else {
+      std::unordered_map<string, int> columns_map;
+      for (int i = 0; i < table_->num_columns(); i++) {
+        columns_map[table_->column(i)->name()] = i;
+      }
+      for (size_t i = 0; i < columns.size(); i++) {
+        columns_index_.push_back(columns_map[columns[i]]);
+      }
+    }
+
+    dtypes_.clear();
+    shapes_.clear();
+    columns_.clear();
+    for (size_t i = 0; i < columns_index_.size(); i++) {
+      int column_index = columns_index_[i];
+      ::tensorflow::DataType dtype;
+      switch (table_->column(column_index)->type()->id()) {
+      case ::arrow::Type::BOOL:
```

Comment:
could you use `arrow::adapters::tensorflow::GetTensorFlowType` here?

Reply:
@BryanCutler It is a little complicated as GetTensorFlowType is in a header file in arrow library. So directly include the header in two `.cc` files will not work. I created a wrapper instead to avoid linking issues.